### ISSUE TITLE
[BUG] reduce_max/min/absmax `nan_propagate=True` is silently ignored for float32/float64

### ISSUE THREAD

**Issue by LeiWang1999** [tile-ai/tilelang#3270]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)
- [x] I have tried the latest version of TileLang.

### What version of TileLang are you using?

main (227515d2)

### System information

Linux, CUDA release 13.1, PyTorch 2.11.0+cu130, NVIDIA H800 (sm_90). The behaviour is decided at lowering time by dtype, not by hardware.

### Problem description

`tilelang.cuda.language.reduce_max/reduce_min/reduce_absmax` accept a `nan_propagate` keyword (added in #2788 and the PR before it). For `float16`/`bfloat16` it lowers the reduction to `__hmax_nan`/`__hmin_nan` and NaNs propagate as documented. For `float32` and `float64` the flag is **silently dropped**: the lowering is identical to `nan_propagate=False`, the reduction still uses `fmaxf`/`fminf` (IEEE `maxNum`, returns the non-NaN operand), and the NaN is swallowed. There is no warning and no error — the user asked for NaN propagation, got `True` accepted, and received a finite result.

Where this happens:

- `src/backend/common/op/reduce.h:276` (and `:327`, `:622`): `use_nan_op = op.nan_propagate && (dtype.is_float16() || dtype.is_bfloat16())` — every other dtype falls through to the plain op.
- `src/tl_templates/cuda/reduce.h`: `MaxOpNan`/`MinOpNan` only specialise `half_t`/`bfloat16_t`; the generic overload is `tl::fast_max`, and `fast_max<float>` is `fmaxf` (`src/tl_templates/cuda/common.h:1334`), so even the "NaN" functor does not propagate for fp32.
- The docstring in `tilelang/cuda/language/reduce_op.py` says the knob "is meaningful for float16/bfloat16 only", and the `ReduceOpNode` field comment says the same — but that is a documented silent no-op, and fp32 is the most common accumulator dtype for exactly the reductions (softmax `m_i`, absmax scaling) where NaN handling matters.

Measured on main with a 128-element row containing one NaN:

| dtype | `nan_propagate=False` | `nan_propagate=True` | `torch.max` |
|---|---|---|---|
| float16 | 127.0 | **nan** | nan |
| bfloat16 | 127.0 | **nan** | nan |
| float32 | 127.0 | 127.0 | nan |
| float64 | 127.0 | 127.0 | nan |

### Reproducible example code

```python
import torch, tilelang, tilelang.language as T
from tilelang.cuda.language import reduce_max

def kernel(dtype, nan_propagate):
    @T.prim_func
    def main(A: T.Tensor((1, 128), dtype), B: T.Tensor((1,), dtype)):
        with T.Kernel(1, threads=128):
            A_local = T.alloc_fragment((1, 128), dtype)
            B_local = T.alloc_fragment((1,), dtype)
            T.copy(A, A_local)
            reduce_max(A_local, B_local, dim=1, nan_propagate=nan_propagate)
            T.copy(B_local, B)
    return main

for dtype in ("float16", "float32"):
    k = tilelang.compile(kernel(dtype, nan_propagate=True))
    a = torch.arange(128, device="cuda", dtype=torch.float32); a[5] = float("nan")
    a = a.to(getattr(torch, dtype)).reshape(1, 128)
    b = torch.empty(1, device="cuda", dtype=a.dtype)
    k(a, b)
    print(dtype, "nan_propagate=True ->", b.float().item())   # float16 -> nan, float32 -> 127.0
```

### Traceback

No traceback; the kernel compiles and runs, returning `127.0` for `float32`.

### Expected behavior

Either of these would be acceptable; the current silent acceptance is not:

1. **Honour the flag for fp32/fp64.** CUDA has no `fmaxf` variant that propagates NaN, but PTX has `max.NaN.f32` / `min.NaN.f32` (sm_80+), and an `isnan`-guarded select works everywhere else. That would give `MaxOpNan`/`MinOpNan` real `float`/`double` overloads.
2. **Reject it loudly.** If fp32/fp64 support is out of scope, `ReduceOpNode` should raise for `nan_propagate=True` on a dtype it cannot honour, the way it already raises for non-CUDA targets (`src/backend/common/op/reduce.h:754`).

### Additional context

Found while reviewing #3205 (`T.clamp` NaN propagation, #3024), which hits the same root cause from the other side: TileLang has no NaN-propagating `max`/`min` primitive for fp32, so every path that needs one (`clamp`, scalar `T.max`/`T.min`, fp32 reductions) either falls back to a select — which the vectorizer cannot vectorize because the condition varies per lane — or silently does not propagate. A single target-lowered NaN-propagating max/min intrinsic (PTX `max.NaN` on CUDA, select fallback elsewhere) would fix this issue, `nan_propagate` for fp32, and the vectorization cost in #3205 together. Related: #2882 (`T.max`/`T.min` constant folding is order-dependent on NaN).
